### PR TITLE
[Merged by Bors] - feat: forward port AlgebraicTopology.DoldKan.Compatibility

### PR DESCRIPTION
**Pull request by joelriou**
This PR is a forward port of mathlib3 PR !3#17923

---
See https://github.com/leanprover-community/mathlib/pull/17923 and https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/!3.2317923

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean [hunk 1]
```diff
@@ -184,6 +184,104 @@ theorem equivalence_functor : (equivalence hF hG).functor = F ⋙ eB.inverse :=
...
+#align algebraic_topology.dold_kan.compatibility.υ AlgebraicTopology.DoldKan.Compatibility.υ
+
+variable (ε : eA.functor ≅ F ⋙ e'.inverse) (hε : υ hF = ε) (hG)
+
+/-- The unit isomorphism of `equivalence`. -/
+@[simps!]
+def equivalenceUnitIso : 𝟭 A ≅ (F ⋙ eB.inverse) ⋙ G :=
+  calc
+    𝟭 A ≅ eA.functor ⋙ eA.inverse := eA.unitIso
+    _ ≅ (F ⋙ e'.inverse) ⋙ eA.inverse := (isoWhiskerRight ε _)
```
Comment (TwoFX) [inline Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean:261]: Just a suggestion:
```suggestion
    _ ≅ (F ⋙ e'.inverse) ⋙ eA.inverse := isoWhiskerRight ε _
```

**Review by TwoFX — APPROVED** [review]
bors d+

File: Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/AlgebraicTopology/DoldKan/Compatibility.lean]: Thanks! I have removed many other unnecessary parentheses.

**Comment by joelriou** [comment]
bors merge

**Pull request closed without merging**